### PR TITLE
drafter 3.2.7 (new formula)

### PR DESCRIPTION
**Pull request by SMillerDev**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This packages drafter, which has a formula in their repo which is pretty much always outdated.
Hopefully this will make the homebrew install a lot easier.

### CODE REVIEW

File: Formula/drafter.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  desc "Snow Crash parser harness"
+  homepage "https://apiblueprint.org"
+  url "https://github.com/apiaryio/drafter/releases/download/v3.2.7/drafter-v3.2.7.tar.gz"
+  sha256 "a2b7061e2524804f153ac2e80f6367ae65dfcd367f4ee406eddecc6303f7f7ef"
+  head "https://github.com/apiaryio/drafter.git"
+
+  def install
+    system "./configure", "--shared"
+    system "make", "drafter"
+    bin.install "bin/drafter"
```
Comment (fxcoudert) [inline Formula/drafter.rb]: `make install`

File: Formula/drafter.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    (include + "drafter").install "src/drafter.h"
+    lib.install "build/out/Release/libdrafter.dylib"
```
Comment (fxcoudert) [inline Formula/drafter.rb]: Please report to upstream the fact that `make install` does not install shared library and header, as it should.

**Review by fxcoudert — DISMISSED** [review]

File: Formula/drafter.rb (hunk 2 above)
Comment (SMillerDev) [reply Formula/drafter.rb]: I found this one already open: https://github.com/apiaryio/drafter/issues/426 is that specific enough?

File: Formula/drafter.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
+class Drafter < Formula
```
Comment (ilovezfs) [inline Formula/drafter.rb]: name is pretty generic

File: Formula/drafter.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  desc "Snow Crash parser harness"
```
Comment (ilovezfs) [inline Formula/drafter.rb]: what does that mean

File: Formula/drafter.rb [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  homepage "https://apiblueprint.org"
```
Comment (ilovezfs) [inline Formula/drafter.rb]: trailing slash

File: Formula/drafter.rb [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  url "https://github.com/apiaryio/drafter/releases/download/v3.2.7/drafter-v3.2.7.tar.gz"
+  sha256 "a2b7061e2524804f153ac2e80f6367ae65dfcd367f4ee406eddecc6303f7f7ef"
+  head "https://github.com/apiaryio/drafter.git"
```
Comment (ilovezfs) [inline Formula/drafter.rb]: see audit

File: Formula/drafter.rb [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+  def install
+    system "./configure", "--shared"
+    system "make", "install", "DESTDIR=#{prefix}"
+    bin.install "bin/drafter"
+    (include + "drafter").install "src/drafter.h"
+    lib.install "build/out/Release/libdrafter.dylib"
```
Comment (ilovezfs) [inline Formula/drafter.rb]: are you actually planning on using this? if make install doesn't handle it, and it's not default, I'd ditch it

File: Formula/drafter.rb [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+  test do
+    (testpath/"api.apib").write <<~EOS
+      # Homebrew API [/brew]
+
+      ## Retrieve All Formula [GET /Formula]
+      + Response 200 (application/json)
+        + Attributes (array)
+    EOS
+    assert_match "\nOK.", shell_output("#{bin}/drafter -l api.apib 2>&1")
```
Comment (ilovezfs) [inline Formula/drafter.rb]: use chomp not \n

Comment (ilovezfs) [reply Formula/drafter.rb]: can it use assert equal?

**Review by ilovezfs — DISMISSED** [review]

File: Formula/drafter.rb (hunk 3 above)
Comment (SMillerDev) [reply Formula/drafter.rb]: Should I prefix it with something?

File: Formula/drafter.rb (hunk 4 above)
Comment (SMillerDev) [reply Formula/drafter.rb]: It's the description of the source repo.

**Comment by SMillerDev** [comment]
@fxcoudert @ilovezfs did these last updates fix the issues you noticed?

File: Formula/drafter.rb (hunk 4 above)
Comment (ilovezfs) [reply Formula/drafter.rb]: I'm still not sure what it means. What is this?

Comment (ilovezfs) [reply Formula/drafter.rb]: https://apiblueprint.org/tools.html says "Native C/C++ API Blueprint Parser" which is a little more intelligible

File: Formula/drafter.rb (hunk 3 above)
Comment (ilovezfs) [reply Formula/drafter.rb]: I guess we can let it land grab.

File: Formula/drafter.rb [hunk 9]
```diff
@@ -0,0 +1,24 @@
+class Drafter < Formula
+  desc "Snow Crash parser harness"
+  homepage "https://apiblueprint.org/"
+  url "https://github.com/apiaryio/drafter/releases/download/v3.2.7/drafter-v3.2.7.tar.gz"
+  sha256 "a2b7061e2524804f153ac2e80f6367ae65dfcd367f4ee406eddecc6303f7f7ef"
+
+  def install
+    system "./configure", "--shared"
+    system "make", "install", "DESTDIR=#{prefix}"
+    bin.install "bin/drafter"
```
Comment (ilovezfs) [inline Formula/drafter.rb]: This line is redundant because it's already been installed by the `make install` above.

File: Formula/drafter.rb [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+    (include + "drafter").install "src/drafter.h"
```
Comment (ilovezfs) [inline Formula/drafter.rb]: don't think we need to install the header at all

File: Formula/drafter.rb (hunk 4 above)
Comment (SMillerDev) [reply Formula/drafter.rb]: Snow Crash is the reference parser (library) for API Blueprint, and this tool is a wrapper around that that makes it into a CLI tool and exposes header files and whatnot.

File: Formula/drafter.rb (hunk 9 above)
Comment (ilovezfs) [reply Formula/drafter.rb]: ping

**Comment by ilovezfs** [comment]
Thanks for the 🆕 formula @SMillerDev! 🚀

**Comment by SMillerDev** [comment]
You're very welcome, thanks for the merge! 🥇

**Pull request closed without merging**